### PR TITLE
Fix some small things in the structure module

### PR DESCRIPTION
**Pull request by hkrutzer**

### CODE REVIEW

File: lib/ecto/adapters/clickhouse/structure.ex [hunk 1]
```diff
@@ -67,6 +65,9 @@ defmodule Ecto.Adapters.ClickHouse.Structure do
       rows = rows |> IO.iodata_to_binary() |> String.replace("),(", "),\n(")
       versions = ["INSERT INTO ", table, " (version, inserted_at) VALUES\n", rows, ";\n"]
       {:ok, versions, conn}
+    else
+      {:ok, _, conn} ->
+        {:ok, "", conn}
```
Comment (ruslandoga) [inline lib/ecto/adapters/clickhouse/structure.ex:70]: Why is this needed? When is the result not `%{rows: rows}`?

File: lib/ecto/adapters/clickhouse/structure.ex [hunk 2]
```diff
@@ -15,8 +15,6 @@ defmodule Ecto.Adapters.ClickHouse.Structure do
     end
   end
 
-  # TODO include views
-
```
Comment (ruslandoga) [inline lib/ecto/adapters/clickhouse/structure.ex:19]: Is it not possible to get views info? https://clickhouse.com/docs/en/sql-reference/statements/create/view

File: lib/ecto/adapters/clickhouse/structure.ex [hunk 3]
```diff
@@ -96,7 +97,7 @@ defmodule Ecto.Adapters.ClickHouse.Structure do
 
   defp run_with_cmd(cmd, cmd_args, opts) do
     args = ["--host", opts[:hostname] || "localhost"]
-    args = if username = opts[:username], do: ["--username", username | args], else: args
+    args = if username = opts[:username], do: ["--user", username | args], else: args
```
Comment (ruslandoga) [inline lib/ecto/adapters/clickhouse/structure.ex:100]: ❤️

File: lib/ecto/adapters/clickhouse/structure.ex (hunk 2 above)
Comment (ruslandoga) [reply lib/ecto/adapters/clickhouse/structure.ex:19]: Ah, I saw you commit messages.

File: lib/ecto/adapters/clickhouse/structure.ex (hunk 1 above)
Comment (hkrutzer) [reply lib/ecto/adapters/clickhouse/structure.ex:70]: I could take this out, this is kind of a niche case. I am migrating from a different Clickhouse client so I want to create a structure dump. Because it's not Ecto, there is no `schema_migrations` table at all. I manually created the table, but it doesn't contain any records yet. In this case, the dump command produces `INSERT INTO schema_migrations VALUES ;`

Comment (ruslandoga) [reply lib/ecto/adapters/clickhouse/structure.ex:70]: I see. But I don't think this change actually fixes the issue. We'd need to add a check on rows being non-empty for this  `else` clause to be matched.

Comment (hkrutzer) [reply lib/ecto/adapters/clickhouse/structure.ex:70]: You are right, I made a mistake with committing. I changed it to what I meant.

File: lib/ecto/adapters/clickhouse/structure.ex [hunk 4]
```diff
@@ -61,10 +61,13 @@ defmodule Ecto.Adapters.ClickHouse.Structure do
     table = @conn.quote_table(database, table)
     stmt = "SELECT * FROM #{table} FORMAT Values"
 
-    with {:ok, %{rows: rows}, conn} <- exec(conn, stmt) do
+    with {:ok, %{rows: [rows]}, conn} <- exec(conn, stmt) do
```
Comment (ruslandoga) [inline lib/ecto/adapters/clickhouse/structure.ex:64]: Technically we can get more than one row in results for tables with lots of rows in schema_migrations table. I think ClickHouse would start splitting it at ~50k characters, it's a 

We'd also need to handle the error case explicitly since we are adding an `else`.

```elixir
else
  {:ok, %{rows: []}, conn} -> []
  {:error, _reason} = error -> error # or was it {:error, _reason, conn}, I don't remember
end
```

Comment (hkrutzer) [reply lib/ecto/adapters/clickhouse/structure.ex:64]: I don't think it is a problem if there are multiple rows:
```
iex(1)> with [x] <- [1,2,3,4] do
...(1)>   x
...(1)> end
[1, 2, 3, 4]
iex(2)>
```

Comment (ruslandoga) [reply lib/ecto/adapters/clickhouse/structure.ex:64]:
```elixir
iex(1)> with [x] <- [1,2,3], do: true
[1, 2, 3]
```

Comment (ruslandoga) [reply lib/ecto/adapters/clickhouse/structure.ex:64]: In `with`, there is an implicit `else` block that returns the unmatched values.

**Comment by hkrutzer** [comment]
I've removed the commit handling the empty `schema_migrations`, it doesn't seem worth it over manually removing the line.

File: lib/ecto/adapters/clickhouse/structure.ex [hunk 5]
```diff
@@ -96,7 +94,7 @@ defmodule Ecto.Adapters.ClickHouse.Structure do
 
   defp run_with_cmd(cmd, cmd_args, opts) do
     args = ["--host", opts[:hostname] || "localhost"]
-    args = if username = opts[:username], do: ["--username", username | args], else: args
+    args = if username = opts[:username], do: ["--user", username | args], else: args
```
Comment (ruslandoga) [inline lib/ecto/adapters/clickhouse/structure.ex:97]: https://github.com/plausible/ecto_ch/pull/111 probably obsoletes this change

Comment (hkrutzer) [reply lib/ecto/adapters/clickhouse/structure.ex:97]: Indeed, so nothing really remains here 🙂
